app/cp_nigeria/helpers.py

```diff
@@ -258,22 +259,43 @@ def get_demand_indicators(project, with_timeseries=False):
 
     :param with_timeseries: when True return the full timeseries as well
     """
-    qs_demand = Asset.objects.filter(scenario=project.scenario, asset_type__asset_type="demand")
+    qs_demand = Asset.objects.filter(scenario=project.scenario, asset_type__asset_type="reducable_demand")
     if qs_demand.exists():
-        demand = json.loads(qs_demand.get().input_timeseries)
-        demand_np = np.array(demand)
-
+        demand_np = []
+        for dem in qs_demand:
+            demand = json.loads(dem.input_timeseries)
+            demand_np.append(demand)
+        demand_np = np.vstack(demand_np).sum(axis=1)
     else:
         demand_np = np.array(get_aggregated_demand(project))
 
     total_demand = demand_np.sum()
     peak_demand = round(demand_np.max(), 1)
     daily_demand = round(total_demand / 365, 1)
     if with_timeseries is True:
-        return (demand, total_demand, peak_demand, daily_demand)
+        return (demand_np.tolist(), total_demand, peak_demand, daily_demand)
     else:
         return (total_demand, peak_demand, daily_demand)
 
+def get_fulfilled_demand_indicator(project, total_only=False):
+    """Provide the aggregated, peak and daily averaged fulfilled demand
+
+    :param total_only: when True return only the aggragated value over the simulation time
+    """
+    qs_res = FancyResults.objects.filter(
+        simulation__scenario=project.scenario, direction="out", bus="ac_bus", asset__contains="critical"
+    )
```

Comment:
I'm a bit confused about this queryset, it is throwing an error for me because it is returning an empty queryset. Is it supposed to be `asset_type__contains="reducable"` instead?

Reply:
Also, this will throw an error for all old projects. Maybe we can get the single demand `if not qs_res.exists()`, or we just leave it since we keep inevitable breaking old projects anyways 😬

Reply:
For you it is empty because you have no simulation results, the oemof sinks created for the reduced demand are called `critical_demand` and `non_critical_demand`, this is why I filter with critical. I will add a `qs_res.exists()`